### ISSUE TITLE
Add possibility to populate serials and UUID on Dell OptiPlex/Precision

### ISSUE THREAD

**Issue by miczyg1** [Dasharo/dasharo-issues#155]
**The problem you're addressing (if any)**
Generation/recovery of unique identification data in SMBIOS.

**Describe the solution you'd like**
Use SERIAL_NUMBER_IN_CBFS feature in coreboot. Deliver UUID from service tag as serial. SMBIOS type 1 serial number is equal to the service tag on the sticker. SMBIOS type 1 UUID is a fixed recipe delivered from service tag. SMBIOS type 2 product name and serial number has not been yet deciphered. SMBIOS type 3 chassis serial number is equal to service tag. Examples:

Dell OptiPlex 7010 service tag 5Q4Q7Y1:
```
Handle 0x0001, DMI type 1, 27 bytes
System Information
	Manufacturer: Dell Inc.
	Product Name: OptiPlex 7010
	Version: 01
	Serial Number: 5Q4Q7Y1
	UUID: 4c4c4544-0051-3410-8051-b5c04f375931
	
Handle 0x0002, DMI type 2, 15 bytes
Base Board Information
	Manufacturer: Dell Inc.
	Product Name: 0WR7PY
	Version: A01
	Serial Number: /5Q4Q7Y1/CN7220036H04QB/
```

UUID format:
1. `4c4c4544` == `DELL` after dehexlify, take endianess into account 
2. `0051` - `0x00` NULL character for `DELL` termination, `0x51` == `Q` is the second letter of the service tag, hexlified
3. `3410` - `0x34` == `4` 3rd letter of the service tag, hexlified; `0x10` - fixed number, present in UUID for Dell Optiplex 7010/9010 and Precision T1650
4. `8051` - `0x80` - fixed number, present in UUID for Dell Optiplex 7010/9010 and Precision T1650, `0x51` == `Q` fourth letter of the service tag, hexlified
5. `b5c04f375931`
   - `0xb5` == `5` | `0x80` first service tag letter, hexlified and ORed with `0x80`; 
   - `c04f`  - fixed number, present in UUID for Dell Optiplex 7010/9010 and Precision T1650
   - `375931`== `7Y1` last three letters of the service tag, hexlified
   
SMBIOS type 2 product name: `0WR7PY` - **need more sample outputs from the machine to confirm it is fixed value**
SMBIOS type 2 product name: `/5Q4Q7Y1/CN7220036H04QB/` - first part is the service tag, second part haven't been discovered what it is yet, also **ned more samples**

Example for OptiPlex 9010: Service tag 9D813W1 , UUID 4C4C4544-0044-3810-8031-B9C04F335731 (same format as above), Product name 051FJ8, serial number /9D813W1/CN722002A802M6/
Example for Precision T1650: Service tag GJBDZY1, UUID 4c4c4544-004a-4210-8044-c7c04f5a5931  (same format as above), Product name 0C3YXR, serial number /GJBDZY1/CN7220037R02C2/

**Where is the value to a user, and who might that user be?**
Uniquely identify the board. May be needed when migrating from vendor firmware, .e.g if SMBIOS data is used by Windows etc.

**Describe alternatives you've considered**
None

**Additional context**
None

**Comment by pietrushnic** [comment]
@miczyg1 what would be Dasharo way of being able to reconstruct that? @macpijan I think DTS should have some application that does it. I know now we are more scripting, but trend seem to be that we will put more and more in DTS. Do you think we should migrate to Rust/C/Go? Also it would be great to have some product plan for DTS.

**Comment by miczyg1** [comment]
We just need to either retrieve the serials from dmidecode during deployment or use the service tag sticker to put serial to CBFS. The rest of the magic should happen in the mainboard code in coreboot.

**Comment by BeataZdunczyk** [comment]
@pietrushnic No, we don't. I've moved it to the: https://github.com/Dasharo/dasharo-issues/milestone/50

**Comment by pietrushnic** [comment]
@BeataZdunczyk thanks.
